src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,209 @@
+        }
+
+        // Build arguments (if any): description and command option (at/by) information (only 1 is expected)
+        String commandOption = "";
+        StringBuilder sbArgument = new StringBuilder();
+        for (int i = 1; i < splitInput.length; i++) {
+            boolean isFirstCommandOption = commandOption.isEmpty() && splitInput[i].startsWith("/");
+            if (isFirstCommandOption) {
+                String description = sbArgument.toString().stripTrailing();
+                commandOption = splitInput[i];
+                inputArguments.add(description);
+                inputArguments.add(commandOption);
+                sbArgument.setLength(0);
+                continue;
+            }
+
+            sbArgument.append(splitInput[i]).append(" ");
+        }
+
+        // The last argument could be a description, task index, at/by information or empty string
+        String lastArgument = sbArgument.toString().stripTrailing();
+        inputArguments.add(lastArgument);
+        return inputArguments.toArray(new String[0]);
+    }
+
+    private static Command generateCommand(String[] inputArguments) throws DukeException {
+        String userCommand = inputArguments[0];
+        String description = inputArguments[1];
+        switch (userCommand) {
+        case COMMAND_BYE:
+            validateCommand(REQUIRED_BYE_ARGUMENT_COUNT, inputArguments);
+            return new ByeCommand();
+        case COMMAND_LIST:
+            validateCommand(REQUIRED_LIST_ARGUMENT_COUNT, inputArguments);
+            return new ListCommand();
+        case COMMAND_DONE:
+            validateCommand(REQUIRED_DONE_ARGUMENT_COUNT, inputArguments);
+            return new DoneCommand(description);
+        case COMMAND_DELETE:
+            validateCommand(REQUIRED_DELETE_ARGUMENT_COUNT, inputArguments);
+            return new DeleteCommand(description);
+        case COMMAND_TODO:
+            validateCommand(REQUIRED_TODO_ARGUMENT_COUNT, inputArguments);
+            Todo newTodo = new Todo(description);
+            return new AddCommand(newTodo);
+        case COMMAND_DEADLINE:
+            validateCommand(REQUIRED_DEADLINE_ARGUMENT_COUNT, inputArguments);
+            String byInformation = inputArguments[3];
+            Deadline newDeadline = new Deadline(description, byInformation);
+            return new AddCommand(newDeadline);
+        case COMMAND_EVENT:
+            validateCommand(REQUIRED_EVENT_ARGUMENT_COUNT, inputArguments);
+            String atInformation = inputArguments[3];
+            Event newEvent = new Event(description, atInformation);
+            return new AddCommand(newEvent);
+        default:
+            throw new DukeException(" Command \"" + userCommand + "\" is not recognized by Duke :(");
+        }
+    }
+
+    private static void validateCommand(int requiredArgumentCount, String[] inputArguments) throws DukeException {
```

Comment:
You may consider abstracting the body of each case out as a helper method to reduce the length of this method and practise SLAP.